content/essentials/list-rendering.mdx

```diff
@@ -0,0 +1,213 @@
+              </For>
+            </ul>
+          </li>
+        )}
+      </For>
+    </ul>
+  );
+}
+```
+
+Through this nested looping structure, the arrangement of `<For>` components provides a way to efficiently render hierarchical data. In this example, a list of categories each containing a list of items is organized and rendered in a clear and structured format.
+
+## `Index`
+
+The `<Index>` component is designed to optimize list rendering in Solid, particularly when working with *primitives*.
+This component provides a way to manage the rendering of lists, ensuring fewer re-renders compared to the `<For>` component under *certain conditions*.
+`<Index>` becomes beneficial when the order and length of the list remain *stable*, but the content may change frequently.
+
+To use `<Index>`, import it directly from Solid:
+
+```jsx
+import { Index } from "solid-js";
+```
+
+Similar to the `<For>` component, `<Index>` accepts a single property named each, which is where you pass the array or object you want to loop over. 
+In the rendering function, JSX elements will represent each item in the DOM, and the `item` and `index` arguments can dynamically set attributes or content of the JSX elements:
+
+```jsx
+<Index each={data()}>
+  {(item, index) => 
+    <li classList={{ even: index % 2 === 0, odd: index % 2 !== 0 }}>
+      {item.name}
+    </li>
+  }
+</Index>
+```
+
+### `<Index>` vs `<For>`
+
+`<Index>` distinguishes itself from the `<For>` component, in that causes *fewer* re-renders. 
+It is concerned primarily with the **index** of the elements in the array, thus allowing the *content* at each index to change without a re-render.
+
+In `<For>`, however, when an array undergoes a change, such as an element shifting position, it will manage this by simply **moving** the corresponding DOM node and **updating** the index.
+This causes a re-render of the entire list, which could impact performance, especially in cases of large or frequently updated lists.
+
+In cases where **primitives** or **nested arrays** are used, it's recommended to use `<Index>` to mitigate any unnecessary renders. 
+Using `<Index>` will ensure that only the content at the *altered* index is updated, providing a more efficient approach to managing dynamic lists.
+
+Consider this revised version of the example in the `<For>` section:
+
+```jsx
+function App() {
+  const categories = [
+    {
+      name: 'Fruits',
+      items: [{ name: 'Apple', selected: false }, { name: 'Banana', selected: false }]
+    },
+    {
+      name: 'Vegetables',
+      items: [{ name: 'Carrot', selected: false }, { name: 'Potato', selected: false }]
+    }
+    // ... more categories
+  ];
+
+  const toggleSelection = () => {
+    // function logic to toggle item selection
+  };
+
+  return (
+    <ul>
+      <For each={categories}>
+        {(category, categoryIndex) => (
+          <li>
+            {category.name}
+            <ul>
+              <Index each={category.items}>
```

Comment:
Point this out with a comment. I missed it at first glance. `// 👈 ` should suffice or highlight the line, you can use a syntax like `jsx {1,9-12} `